libpod/healthcheck_linux.go

```diff
@@ -152,10 +172,13 @@ func (c *Container) disableHealthCheckSystemd(isStartup bool) bool {
 }
 
 // Systemd unit name for the healthcheck systemd unit
-func (c *Container) hcUnitName(isStartup bool) string {
+func (c *Container) hcUnitName(isStartup, bare bool) string {
 	unitName := c.ID()
 	if isStartup {
 		unitName += "-startup"
 	}
+	if !bare {
+		unitName += fmt.Sprintf("-%x", rand.Int())
```

Comment:
should we use crypto/rand instead?

Reply:
Does it matter? We just need to make sure that we don't repeat within a ~reasonable period of iterations, and the normal RNG is more than enough for that.

Reply:
it used to be the fact the math/rand has to be seeded first, I don't think that longer holds as of go 1.21 so we are likely fine but I know we had bugs because of that before...
We use cyrpto/rand to generate untis names elsewhere already:
https://github.com/containers/common/blob/6b42205e407e56fe0e1b1226d684d2b7e01f9c5c/pkg/systemd/systemd_linux.go#L64-L66

So this is a known quantity to me that I can trust to work

Reply:
How firm are you on this desire? Because crypto/rand can return errors, which would mean signature changes for this function to return that error, and I don't feel like it's worth the change.

Reply:
I am fine keeping rand.Int(), looks like the auto seed was added in go 1.20 so we should be safe in all cases as we require go 1.21 today